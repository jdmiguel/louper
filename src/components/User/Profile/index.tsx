import Typography from '@mui/material/Typography';
import Avatar from '@mui/material/Avatar';
import TextTag from '@/components/shared/TextTag';
import Link from '@/components/shared/Link';
import { ICON_TYPE, LINK_TEXT } from '@/utils/literals';
import { UserData } from '@/utils/types';
import {
  StyledRoot,
  StyledAvatarWrapper,
  StyledContent,
  StyledHeadings,
  StyledDetails,
  StyledTextTagWrapper,
  StyledAction,
} from './styles';

const buildCreationDate = (date: string) => {
  const newDate = new Date(date);
  const formattedDate = newDate.toLocaleString();

  return formattedDate.slice(0, formattedDate.indexOf(','));
};

type Props = {
  userData: UserData;
};

const Profile = ({
  userData: {
    avatar_url,
    name,
    login,
    bio,
    public_repos,
    following,
    followers,
    created_at,
    location,
    company,
    html_url,
    email,
    blog,
  },
}: Props) => (
  <StyledRoot>
    <StyledAvatarWrapper>
      <Avatar sx={{ width: 240, height: 240, position: 'absolute' }} />
      <Avatar
        alt="user avatar"
        src={avatar_url}
        sx={{
          width: 240,
          height: 240,
          position: 'absolute',
          boxShadow: '0px 0px 14px 6px rgb(169 45 201 / 40%)',
        }}
      />
    </StyledAvatarWrapper>
    <StyledContent>
      <StyledHeadings>
        {name && (
          <Typography variant="h3" sx={{ marginBottom: 0.4 }}>
            {name}
          </Typography>
        )}
        <Typography variant="h4" sx={{ marginBottom: 2 }}>
          {login}
        </Typography>
        {bio && (
          <Typography
            sx={{
              color: 'secondary.light',
              overflow: 'hidden',
              maxWidth: 240,
              textOverflow: 'ellipsis',
            }}
          >
            {bio}
          </Typography>
        )}
      </StyledHeadings>
      <StyledDetails>
        <StyledTextTagWrapper>
          <TextTag content={`${public_repos}`} withUppercase withIcon iconType={ICON_TYPE.repo} />
        </StyledTextTagWrapper>
        <StyledTextTagWrapper>
          <TextTag content={`${following}`} withUppercase withIcon iconType={ICON_TYPE.following} />
        </StyledTextTagWrapper>
        <StyledTextTagWrapper>
          <TextTag content={`${followers}`} withUppercase withIcon iconType={ICON_TYPE.followers} />
        </StyledTextTagWrapper>
        <StyledTextTagWrapper>
          <TextTag
            content={buildCreationDate(created_at)}
            withIcon
            iconType={ICON_TYPE.creationDate}
          />
        </StyledTextTagWrapper>
        {location && (
          <StyledTextTagWrapper>
            <TextTag content={location} withIcon iconType={ICON_TYPE.location} />
          </StyledTextTagWrapper>
        )}
        {company && (
          <StyledTextTagWrapper>
            <TextTag content={company} withIcon iconType={ICON_TYPE.company} />
          </StyledTextTagWrapper>
        )}
      </StyledDetails>
      <>
        <StyledAction>
          <Link
            url={html_url}
            ariaLabel={`View ${login} profile on GitHub`}
            content={LINK_TEXT.profile}
            withIcon
            iconType={ICON_TYPE.profile}
          />
        </StyledAction>
        {email && (
          <StyledAction>
            <Link
              url={`mailto:${email}`}
              ariaLabel={`Send email to ${login}`}
              content={LINK_TEXT.email}
              withIcon
              iconType={ICON_TYPE.email}
            />
          </StyledAction>
        )}
        {blog && (
          <StyledAction>
            <Link
              url={blog}
              ariaLabel={`View portfolio of ${login}`}
              content={LINK_TEXT.portfolio}
              withIcon
              iconType={ICON_TYPE.portfolio}
            />
          </StyledAction>
        )}
      </>
    </StyledContent>
  </StyledRoot>
);

export default Profile;
