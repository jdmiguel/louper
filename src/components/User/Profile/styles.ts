import { styled } from '@mui/material/styles';

export const StyledRoot = styled('main')({
  display: 'flex',
  flexDirection: 'column',
  maxWidth: 240,
  position: 'sticky',
  top: 60,
});

export const StyledAvatarWrapper = styled('div')({
  height: 240,
  position: 'relative',
  width: 240,
});

export const StyledContent = styled('div')({
  marginTop: 30,
  maxWidth: 300,
  overflow: 'hidden',
});

export const StyledHeadings = styled('div')({
  marginBottom: 24,
});

export const StyledDetails = styled('div')(({ theme }) => ({
  borderBottom: `1px solid ${theme.palette.secondary.light}`,
  borderTop: `1px solid ${theme.palette.secondary.light}`,
  display: 'flex',
  flexWrap: 'wrap',
  marginBottom: 24,
  paddingBottom: 10,
  paddingTop: 10,
}));

export const StyledTextTagWrapper = styled('div')({
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center',
  marginRight: 12,
  marginBottom: 6,
  fontSize: '0.9rem',
});

export const StyledAction = styled('div')({
  marginBottom: 5,
});
