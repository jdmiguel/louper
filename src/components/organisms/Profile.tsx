/* material-ui */
import { styled } from '@mui/material/styles';
import Typography from '@mui/material/Typography';

/* atoms */
import Avatar from '@mui/material/Avatar';
import TextTag from '../atoms/TextTag';
import Link from '../atoms/Link';

/* types */
import { UserData } from '../../utils/types';

const buildCreationDate = (date: string) => {
  const newDate = new Date(date);
  const formattedDate = newDate.toLocaleString();

  return formattedDate.slice(0, formattedDate.indexOf(','));
};

const Root = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  maxWidth: 240,
  position: 'sticky',
  top: 60,
});

const AvatarWrapper = styled('div')({
  height: 240,
  position: 'relative',
  width: 240,
});

const Content = styled('div')({
  marginTop: 20,
  maxWidth: 300,
  overflow: 'hidden',
});

const Info = styled('div')(({ theme }) => ({
  borderBottom: `1px solid ${theme.palette.secondary.light}`,
  borderTop: `1px solid ${theme.palette.secondary.light}`,
  display: 'flex',
  flexWrap: 'wrap',
  marginTop: 24,
  paddingBottom: 10,
  paddingTop: 10,
}));

const TextTagWrapper = styled('div')(({ theme }) => ({
  alignItems: 'center',
  color: theme.palette.secondary.main,
  display: 'flex',
  justifyContent: 'center',
  marginRight: 12,
  marginBottom: 6,
  fontSize: '0.9rem',
}));

const Actions = styled('div')({
  marginTop: 20,
});

const Action = styled('div')({
  marginBottom: 5,
});

type Props = {
  userData: UserData;
};

const Profile = ({
  userData: {
    avatar_url,
    name,
    login,
    bio,
    public_repos,
    following,
    followers,
    created_at,
    location,
    company,
    html_url,
    email,
    blog,
  },
}: Props) => (
  <Root>
    <AvatarWrapper>
      <Avatar sx={{ width: 240, height: 240, position: 'absolute' }} />
      <Avatar
        alt="user avatar"
        src={avatar_url}
        sx={{
          width: 240,
          height: 240,
          position: 'absolute',
          boxShadow: '0 3px 10px rgba(0,0,0,.25)',
        }}
      />
    </AvatarWrapper>
    <Content>
      <Typography variant="h3" sx={{ marginBottom: 0.4 }}>
        {name}
      </Typography>
      <Typography variant="h4" sx={{ marginBottom: 2 }}>
        {login}
      </Typography>
      <Typography
        variant="body2"
        sx={{
          color: 'text.secondary',
          overflow: 'hidden',
          maxWidth: 240,
          textOverflow: 'ellipsis',
        }}
      >
        {bio}
      </Typography>
      <Info>
        <TextTagWrapper>
          <TextTag content={`${public_repos}`} withIcon iconType="folder" />
        </TextTagWrapper>
        <TextTagWrapper>
          <TextTag content={`${following}`} withIcon iconType="visibility" />
        </TextTagWrapper>
        <TextTagWrapper>
          <TextTag content={`${followers}`} withIcon iconType="favorite" />
        </TextTagWrapper>
        {created_at && (
          <TextTagWrapper>
            <TextTag content={buildCreationDate(created_at)} withIcon iconType="event_note" />
          </TextTagWrapper>
        )}
        {location && (
          <TextTagWrapper>
            <TextTag content={location} withLowerCase withIcon iconType="location_on" />
          </TextTagWrapper>
        )}
        {company && (
          <TextTagWrapper>
            <TextTag content={company} withLowerCase withIcon iconType="business" />
          </TextTagWrapper>
        )}
      </Info>
      <Actions>
        <Action>
          <Link
            url={html_url}
            ariaLabel={`View ${name} profile on GitHub`}
            content="VISIT PROFILE"
            withIcon
            iconType="person"
          />
        </Action>
        {email && (
          <Action>
            <Link
              url={`mailto:${email}`}
              ariaLabel={`Send email to ${login}`}
              content="SEND EMAIL"
              withIcon
              iconType="mail_outline"
            />
          </Action>
        )}
        {blog && (
          <Action>
            <Link
              url={blog}
              ariaLabel={`View portfolio of ${name}`}
              content="VISIT PORTFOLIO"
              withIcon
              iconType="web_icon"
            />
          </Action>
        )}
      </Actions>
    </Content>
  </Root>
);

export default Profile;
