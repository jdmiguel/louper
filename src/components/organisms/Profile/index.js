/* material-ui */
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import EventIcon from '@material-ui/icons/Event';
import FolderIcon from '@material-ui/icons/Folder';
import VisibilityIcon from '@material-ui/icons/Visibility';
import FavoriteIcon from '@material-ui/icons/Favorite';
import EmailIcon from '@material-ui/icons/Email';
import LocationOnIcon from '@material-ui/icons/LocationOn';
import AccountBoxIcon from '@material-ui/icons/AccountBox';
import CompanyIcon from '@material-ui/icons/Business';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import Icon from '@material-ui/core/Icon';

/* atoms */
import GithubIcon from '../../atoms/GithubIcon';

/* utils */
import { dataModel } from '../../../utils/models';

const useStyles = makeStyles((theme) => ({
  wrapper: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'center',
    '& > img': {
      borderRadius: '50%',
      height: 240,
      width: 240,
      '@media (max-width: 768px)': {
        height: 180,
        width: 180,
      },
    },
    marginTop: '25vh',
    '@media (max-width: 768px)': {
      marginTop: '12vw',
      flexDirection: 'column',
    },
  },
  profileContent: {
    marginLeft: 30,
    '@media (max-width: 768px)': {
      textAlign: 'center',
      marginLeft: 0,
      marginTop: 36,
    },
  },
  profileGithubInfo: {
    justifyContent: 'space-between',
    alignItems: 'center',
    display: 'flex',
    marginTop: 18,
    marginBottom: 12,
    '@media (max-width: 768px)': {
      alignItems: 'center',
      flexDirection: 'column',
    },
  },
  profileGithubIcon: {
    fontSize: 'initial',
    lineHeight: 0,
    '& svg': {
      width: 22,
      '& path': {
        fill: theme.palette.primary.contrastText,
      },
    },
  },
  profileActions: {
    alignItems: 'center',
    display: 'flex',
    flex: 1,
    marginRight: 14,
    '@media (max-width: 768px)': {
      justifyContent: 'center',
      marginRight: 0,
      marginBottom: 30,
    },
  },
  profileData: {
    alignItems: 'center',
    display: 'flex',
    marginBottom: 2,
  },
  profileTag: {
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
    alignItems: 'center',
    marginRight: 8,
  },
  profileTagIcon: {
    color: theme.palette.text.main,
    fontSize: 25,
    padding: 4,
  },
  profileButton: {
    backgroundColor: theme.palette.primary.main,
    border: 0,
    borderRadius: '50%',
    color: theme.palette.primary.contrastText,
    height: 22,
    padding: 18,
    marginRight: 8,
    width: 22,
    '&:hover': {
      backgroundColor: theme.palette.primary.light,
    },
    '@media (max-width: 768px)': {
      justifyContent: 'center',
    },
  },
  profileIconButton: {
    fontSize: 24,
  },
  gutterBottom: {
    marginBottom: 4,
  },
}));

const formatDate = (date: Date) => {
  const newDate = date.toLocaleString();
  return newDate.slice(0, newDate.indexOf(','));
};

const buildDate = (date: string) => {
  const newDate = new Date(date);
  return formatDate(newDate);
};

const Profile = ({
  data: {
    avatarUrl,
    name,
    bio,
    email,
    repos,
    followers,
    following,
    createdAt,
    location,
    blog,
    company,
    htmlUrl,
  },
}) => {
  const classes = useStyles();
  return (
    <div data-test="profile__wrapper" className={classes.wrapper}>
      <img data-test="profile__image" alt="user avatar" src={avatarUrl} />
      <div className="profile__content" className={classes.profileContent}>
        <Typography
          variant="h3"
          gutterBottom
          classes={{
            gutterBottom: classes.gutterBottom,
          }}
        >
          {name}
        </Typography>
        <Typography variant="h4">{bio}</Typography>
        <div className={classes.profileGithubInfo}>
          <div className={classes.profileData}>
            <div className={classes.profileTag}>
              <FolderIcon
                classes={{
                  root: classes.profileTagIcon,
                }}
                className="profile__icon"
              />
              {repos}
            </div>
            <div className={classes.profileTag}>
              <VisibilityIcon
                classes={{
                  root: classes.profileTagIcon,
                }}
                className="profile__icon"
              />
              {following}
            </div>
            <div className={classes.profileTag}>
              <FavoriteIcon
                classes={{
                  root: classes.profileTagIcon,
                }}
                className="profile__icon"
              />
              {followers}
            </div>
          </div>
          <div className={classes.profileData}>
            {createdAt && (
              <div className={classes.profileTag}>
                <EventIcon
                  classes={{
                    root: classes.profileTagIcon,
                  }}
                  className="profile__icon"
                />
                {buildDate(createdAt)}
              </div>
            )}
            {location && (
              <div className={classes.profileTag}>
                <LocationOnIcon
                  classes={{
                    root: classes.profileTagIcon,
                  }}
                  className="profile__icon"
                />
                {location}
              </div>
            )}
            {company && (
              <div className={classes.profileTag}>
                <CompanyIcon
                  classes={{
                    root: classes.profileTagIcon,
                  }}
                  className="profile__icon"
                />
                {company}
              </div>
            )}
          </div>
        </div>

        <div className={classes.profileActions}>
          <IconButton
            classes={{
              root: classes.profileButton,
            }}
            onClick={() => window.open(htmlUrl, '_blank')}
          >
            <div className={classes.profileGithubIcon}>
              <GithubIcon />
            </div>
          </IconButton>
          {email && (
            <IconButton
              classes={{
                root: classes.profileButton,
              }}
              size="small"
              onClick={() => window.open(`mailto:${email}`, '_blank')}
            >
              <Icon
                classes={{
                  root: classes.profileIconButton,
                }}
              >
                email
              </Icon>
            </IconButton>
          )}
          {blog && (
            <IconButton
              classes={{
                root: classes.profileButton,
              }}
              size="small"
              onClick={() => window.open(blog, '_blank')}
            >
              <Icon
                classes={{
                  root: classes.profileIconButton,
                }}
              >
                link
              </Icon>
            </IconButton>
          )}
        </div>
      </div>
    </div>
  );
};

Profile.propTypes = {
  data: dataModel,
};

export default Profile;
