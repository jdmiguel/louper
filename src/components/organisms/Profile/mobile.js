/* material-ui */
import { styled } from '@mui/material/styles';
import Typography from '@mui/material/Typography';

/* utils */
import { dataModel } from '../../../utils/models';

const Root = styled('div')({
  display: 'flex',
  marginTop: 20,
  '@media (min-width: 769px)': {
    display: 'none',
  },
  '& > img': {
    borderRadius: '50%',
    height: 100,
    marginRight: 20,
    width: 100,
  },
});

const Content = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
});

const ProfileMobile = ({ data: { avatarUrl, name, login } }) => (
  <Root>
    <img alt="user avatar" src={avatarUrl} />
    <Content>
      <Typography variant="h3" sx={{ marginBottom: 0.2 }}>
        {name}
      </Typography>
      <Typography variant="h4">{login}</Typography>
    </Content>
  </Root>
);

ProfileMobile.propTypes = {
  data: dataModel,
};

export default ProfileMobile;
