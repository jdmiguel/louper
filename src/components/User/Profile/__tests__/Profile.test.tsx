import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Profile from '..';

describe('<Profile />', () => {
  const props = {
    userData: {
      login: 'jdmiguel',
      html_url: 'https://github.com/jdmiguel',
      avatar_url: 'https://github.com/jdmiguel.jpg',
      created_at: '2014-03-20T20:24:22Z',
      name: 'Jaime De Miguel',
      bio: 'Frontend developer',
      email: 'jdmiguel@gmail.com',
      location: 'Dublin',
      blog: 'https://jdmiguel.netlify.app/',
      company: 'Kitman Labs',
      public_repos: 30,
      followers: 12,
      following: 16,
    },
  };

  it('displays the correct avatar', () => {
    render(<Profile {...props} />);

    const userAvatar = screen.getByAltText('user avatar');
    expect(userAvatar.getAttribute('src')).toBe('https://github.com/jdmiguel.jpg');
  });

  it('displays the correct main data', () => {
    render(<Profile {...props} />);

    const mainInfo = screen.getByText('jdmiguel').parentElement;
    expect(mainInfo.children).toHaveLength(3);

    expect(screen.getByText('Jaime De Miguel')).toBeInTheDocument();
    expect(screen.getByText('Frontend developer')).toBeInTheDocument();
  });

  it('does not display the name field when no name', () => {
    render(<Profile {...props} userData={{ ...props.userData, name: null }} />);

    const mainInfo = screen.getByText('jdmiguel').parentElement;
    expect(mainInfo.children).toHaveLength(2);

    expect(screen.queryByText('Jaime De Miguel')).not.toBeInTheDocument();
  });

  it('does not display the bio field when no bio', () => {
    render(<Profile {...props} userData={{ ...props.userData, bio: null }} />);

    const mainInfo = screen.getByText('jdmiguel').parentElement;
    expect(mainInfo.children).toHaveLength(2);

    expect(screen.queryByText('Frontend developer')).not.toBeInTheDocument();
  });

  it('renders the correct info details', () => {
    render(<Profile {...props} />);

    const details = screen.getByText('jdmiguel').parentElement.nextElementSibling;
    expect(details.children).toHaveLength(6);

    // repos
    expect(screen.getByText(30)).toBeInTheDocument();
    expect(screen.getByText('folder')).toBeInTheDocument();

    // following
    expect(screen.getByText(16)).toBeInTheDocument();
    expect(screen.getByText('visibility')).toBeInTheDocument();

    // followers
    expect(screen.getByText(12)).toBeInTheDocument();
    expect(screen.getByText('favorite')).toBeInTheDocument();

    // created_at
    expect(screen.getByText('3/20/2014')).toBeInTheDocument();
    expect(screen.getByText('event_note')).toBeInTheDocument();

    // location
    expect(screen.getByText('Dublin')).toBeInTheDocument();
    expect(screen.getByText('location_on')).toBeInTheDocument();

    // company
    expect(screen.getByText('Kitman Labs')).toBeInTheDocument();
    expect(screen.getByText('business')).toBeInTheDocument();
  });

  it('does not display the location field when no location', () => {
    render(<Profile {...props} userData={{ ...props.userData, location: null }} />);

    const details = screen.getByText('jdmiguel').parentElement.nextElementSibling;
    expect(details.children).toHaveLength(5);

    expect(screen.queryByText('Dublin')).not.toBeInTheDocument();
    expect(screen.queryByText('location_on')).not.toBeInTheDocument();
  });

  it('does not display the company field when no location', () => {
    render(<Profile {...props} userData={{ ...props.userData, company: null }} />);

    const details = screen.getByText('jdmiguel').parentElement.nextElementSibling;
    expect(details.children).toHaveLength(5);

    expect(screen.queryByText('Kitman Labs')).not.toBeInTheDocument();
    expect(screen.queryByText('business')).not.toBeInTheDocument();
  });

  it('displays the correct actions', () => {
    render(<Profile {...props} />);

    // profile action
    const profileLink = screen.getByText(/visit profile/i);
    expect(profileLink.getAttribute('href')).toBe('https://github.com/jdmiguel');
    expect(profileLink.getAttribute('aria-label')).toBe('View jdmiguel profile on GitHub');

    // email action
    const emailLink = screen.getByText(/send email/i);
    expect(emailLink.getAttribute('href')).toBe('mailto:jdmiguel@gmail.com');
    expect(emailLink.getAttribute('aria-label')).toBe('Send email to jdmiguel');
    expect(screen.getByText('mail_outline')).toBeInTheDocument();

    // blog action
    const blogLink = screen.getByText(/visit portfolio/i);
    expect(blogLink.getAttribute('href')).toBe('https://jdmiguel.netlify.app/');
    expect(blogLink.getAttribute('aria-label')).toBe('View portfolio of jdmiguel');
    expect(screen.getByText('web_icon')).toBeInTheDocument();
  });

  it('does not display the email action when no email', () => {
    render(<Profile {...props} userData={{ ...props.userData, email: null }} />);

    expect(screen.queryByText(/send email/i)).not.toBeInTheDocument();
    expect(screen.queryByText('mail_outline')).not.toBeInTheDocument();
  });

  it('does not display the blog action when no blog', () => {
    render(<Profile {...props} userData={{ ...props.userData, blog: null }} />);

    expect(screen.queryByText(/visit portfolio/i)).not.toBeInTheDocument();
    expect(screen.queryByText('web_icon')).not.toBeInTheDocument();
  });
});
