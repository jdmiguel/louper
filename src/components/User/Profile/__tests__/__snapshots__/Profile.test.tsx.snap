// Vitest Snapshot v1

exports[`<Profile /> > displays the content properly 1`] = `
<div>
  <main
    class="css-12cao3m"
  >
    <div
      class="css-1wkrglf"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-zsnfb9-MuiAvatar-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1s3wu4-MuiAvatar-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-1v5r5rc"
    >
      <div
        class="css-i3pbo"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-pxw7rc-MuiTypography-root"
        />
      </div>
      <div
        class="css-r8nah5"
      >
        <div
          class="css-x8ggji"
        >
          <div
            class="css-87uc0g"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-r5mrhv-MuiIcon-root"
            >
              folder
            </span>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1o3q8fu-MuiTypography-root"
            >
              0
            </h5>
          </div>
        </div>
        <div
          class="css-x8ggji"
        >
          <div
            class="css-87uc0g"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-r5mrhv-MuiIcon-root"
            >
              visibility
            </span>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1o3q8fu-MuiTypography-root"
            >
              0
            </h5>
          </div>
        </div>
        <div
          class="css-x8ggji"
        >
          <div
            class="css-87uc0g"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-r5mrhv-MuiIcon-root"
            >
              favorite
            </span>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1o3q8fu-MuiTypography-root"
            >
              0
            </h5>
          </div>
        </div>
        <div
          class="css-x8ggji"
        >
          <div
            class="css-87uc0g"
          >
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-r5mrhv-MuiIcon-root"
            >
              event_note
            </span>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1y33nkg-MuiTypography-root"
            >
              Invalid Dat
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-r7q7qr"
      >
        <a
          aria-label="View  profile on GitHub"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1efa7me-MuiTypography-root-MuiLink-root"
          href=""
          rel="noopener noreferrer"
        >
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-7qvvgc-MuiIcon-root"
          >
            person
          </span>
          VISIT PROFILE
        </a>
      </div>
    </div>
  </main>
</div>
`;
